helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -554,12 +554,15 @@ private void subscribeChildChange(String path, NotificationContext.Type callback
       if (!childrenSubscribeResult.isInstalled()) {
         logger.info("CallbackHandler {} subscribe data path {} failed!", this, path);
       }
+      return childrenSubscribeResult.getChildren();
```

Comment:
I think the main issue here is that line 551
`_zkClient.subscribeChildChanges(path, this, callbackType != Type.INIT);` we have two code path.

if callbackType is init, the parent path of callback handler may not be created. Thus, the return would be null, or empty? Please verify.

The potentially issue is that note later at `default`, we re-use the return children list from here. 
Note, original code retrieve children one more time and it may retrieve node by then. Here, it may not.

This is different from previous logic. Not sure if this would bring a bug or not.